manifests/files.pp

```diff
@@ -47,78 +47,75 @@
 define swap_file::files (
   Enum['present', 'absent'] $ensure          = 'present',
   Stdlib::Absolutepath      $swapfile        = '/mnt/swap.1',
-  String                    $swapfilesize    = $facts['memory']['system']['total'],
+  String[1]                 $swapfilesize    = $facts['memory']['system']['total'],
   Boolean                   $add_mount       = true,
-  String                    $options         = 'defaults',
+  String[1]                 $options         = 'defaults',
   Integer                   $timeout         = 300,
-  String                    $cmd             = 'dd',
+  Enum['dd', 'fallocate']   $cmd             = 'dd',
   Boolean                   $resize_existing = false,
-  String                    $resize_margin   = '50MB',
+  String[1]                 $resize_margin   = '50MB',
   Boolean                   $resize_verbose  = false,
 ) {
   $swapfilesize_mb = to_bytes($swapfilesize) / 1048576
 
   if $ensure == 'present' {
-    if ($resize_existing and $facts['swapfile_sizes']) {
-      if (is_hash($facts['swapfile_sizes'])) {
-        if (has_key($facts['swapfile_sizes'],$swapfile)) {
-          swap_file::resize { $swapfile:
-            swapfile_path          => $swapfile,
-            margin                 => $resize_margin,
-            expected_swapfile_size => $swapfilesize,
-            actual_swapfile_size   => $facts['swapfile_sizes'][$swapfile],
-            verbose                => $resize_verbose,
-            before                 => Exec["Create swap file ${swapfile}"],
-          }
-        }
+    # Check for resizing the swap file
+    if $resize_existing and $facts['swapfile_sizes'] {
+      # use $swapfile_sizes for new or $swapfile_sizes_csv as fallback for old Puppet clients
```

Comment:
It feels like something is either missing here is is incorrect. First we're checking if $facts['swapfile_exists'] exists and then on line 65 we grab information from $facts['swapfile_sizes_csv']?

Am I missing something here? Shouldn't we check if swapfile_sizes_csv exists as well or is that part handled by the function that is called?

Reply:
I can only guess here:
both facts `swapfile_sizes` and `swapfile_sizes_csv` are based on the existence of `/proc/swaps`. Either the author is sure both or non will exist or he didn't care for old Puppet clients that much (much like I do :)

We could remove the support for old Puppet clients completely.